examples/http/client.cpp

```diff
@@ -0,0 +1,125 @@
+// Simple HTTP client that prints the response
+
+#include "caf/net/http/client.hpp"
+
+#include "caf/net/http/with.hpp"
+#include "caf/net/ip.hpp"
+#include "caf/net/middleman.hpp"
+#include "caf/net/octet_stream/transport.hpp"
+#include "caf/net/tcp_stream_socket.hpp"
+
+#include "caf/actor_system.hpp"
+#include "caf/actor_system_config.hpp"
+#include "caf/caf_main.hpp"
+#include "caf/detail/latch.hpp"
+#include "caf/event_based_actor.hpp"
+#include "caf/ipv4_address.hpp"
+#include "caf/scheduled_actor/flow.hpp"
+#include "caf/span.hpp"
+#include "caf/uuid.hpp"
+
+#include <cassert>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+using namespace std::literals;
+
+namespace http = caf::net::http;
+namespace ssl = caf::net::ssl;
+using namespace caf;
+
+// -- constants ----------------------------------------------------------------
+
+// static constexpr uint16_t default_port = 7788;
+// static constexpr std::string_view default_host = "localhost";
+
+constexpr auto default_method = http::method::get;
+
+// -- configuration setup ------------------------------------------------------
+struct config : actor_system_config {
+  config() {
+    opt_group{custom_options_, "global"}
+      .add<http::method>("method,m", "HTTP method to use")
+      .add<uri>("resource,r", "Requested resource (URL)")
```

Comment:
I think using a positional argument would be more natural, i.e., `client http://my-host/my-path` (as opposed to `client -r http://my-host/my-path`.